src/Compilers/Core/Portable/Symbols/SymbolEqualityComparer.cs

```diff
@@ -24,6 +24,8 @@ public sealed class SymbolEqualityComparer : IEqualityComparer<ISymbol>
         // Internal only comparisons:
         internal readonly static SymbolEqualityComparer ConsiderEverything = new SymbolEqualityComparer(TypeCompareKind.ConsiderEverything);
 
+        internal readonly static SymbolEqualityComparer IgnoreNullableAndTupleNames = new SymbolEqualityComparer(TypeCompareKind.AllIgnoreOptions | TypeCompareKind.IgnoreTupleNames);
```

Comment:
>IgnoreNullableAndTupleNames [](start = 56, length = 27)

It looks like we already have EqualsIgnoringTupleNamesAndNullability comparer at [https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs#L121](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs#L121).

Reply:
I think that we should avoid using ```IEqualityComparer<ISymbol>``` within compilers for internal symbol classes. This causes an unnecessary overhead in the form of type checks before we call back into the compiler. Second, we are very likely to stop implementing ISymbol on internal symbols soon.

---
In reply to: [327332447](https://github.com/dotnet/roslyn/pull/38769#discussion_r327332447) [](ancestors = 327332447)

Reply:
I'll do this for the set of comparisons introduced in this PR, but I think it's out of scope to change all usages.

---
In reply to: [327335261](https://github.com/dotnet/roslyn/pull/38769#discussion_r327335261) [](ancestors = 327335261,327332447)

Reply:
> I think it's out of scope to change all usages.

Agree. Let's just no make the problem bigger.

---
In reply to: [327350175](https://github.com/dotnet/roslyn/pull/38769#discussion_r327350175) [](ancestors = 327350175,327335261,327332447)